addons/thtracker/thtracker.lua

```diff
@@ -24,70 +24,162 @@
-        th:show()
-
-    else
-        print('th help : Shows help message')
-        print('th pos <x> <y> : Positions the list')
-        print('th hide : Hides the box')
-        print('th show : Shows the box')
-
-    end
+_addon.version = '1.1a'
+
+
+local config = require('config')
+texts = require('texts')
+require('logger')
+
+local thoutput = 'No current mob'
+local posx = 200
+local posy = 200
+local see = true
+local name = nil
+local box_name = 'addon:gr:thbox'
+mob = nil
+count = 0
+
+local settingtab = nil
+local settings_file = 'data\\settings.xml'
+local settingtab = config.load(settings_file)
+if settingtab == nil then
+	write('No settings file found. Ensure you have a file at data\\settings.xml')
+end
+
+if settingtab['posx'] ~= nil then
+	posx = settingtab['posx']
+	posy = settingtab['posy']
+end	
+	
+function initialize()
+	--[[windower.text.create(box_name)
+	windower.text.set_bg_color(box_name,200,30,30,30)
+	windower.text.set_color(box_name,255,200,200,200)
+	windower.text.set_location(box_name,posx,posy)
+	windower.text.set_visibility(box_name, see)
+	windower.text.set_bg_visibility(box_name,1)
+	windower.text.set_font(box_name,'Arial',12)
+	windower.text.set_text(box_name, thoutput)]]
+	--log('Initializing TH Tracker')
+	t = texts.new('test text box')
+	thbox = texts.new('${boxtext}', {
+		pos = {
+			x = posx,
+			y = posy
+		},
+		bg = {
+			visible = true,
+			alpha = 255,
+			red = 30,
+			green = 30,
+			blue = 30
+		},
+		text = {
+			size = 10,
+			alpha = 200,
+			red = 200,
+			green = 200,
+			blue = 200
+		}
+	})
+	thbox.boxtext = thoutput
+	thbox:show()
+end
+
+function dispose()
+    windower.text.delete(box_name)
+end
+
+function thunload()
+	dispose()
+	windower.send_command('unalias th')
+end 
+
+windower.register_event('load', function(...)
+	windower.send_command('alias th lua c thtracker')
+	
+	initialize()
+end)
+
+windower.register_event('login', initialize)
+windower.register_event('logout', dispose)
+windower.register_event('unload', thunload)
+
+
+windower.register_event('addon command', function(...)
+	local params = {...};
+	if #params < 1 then
+		return
+	end
+	if params[1] then
+		if params[1]:lower() == "help" then
+			print('th help : Shows help message')
+			print('th pos <x> <y> : Positions the list')
+			print('th hide : Hides the box')
+			print('th show : Shows the box')
+		elseif params[1]:lower() == "pos" then
+			if params[3] then
+				thbox.posx(tonumber(params[2]))
+				thbox.posy(tonumber(params[3]))
+				thbox:update()
+				--windower.text.set_location(box_name, posx, posy)
+			else
+				print('THTracker is currently at '..posx..', '..posy)
+			end
+		elseif params[1]:lower() == "hide" then
+			see = false
+			thbox:hide()
+		elseif params[1]:lower() == "show" then
+			see = true
+			thbox:show()
+		elseif params[1]:lower() == "current" then
+			print('Box text should be: '..thoutput)
+		elseif params[1]:lower() == "delete" then
+			--windower.text.delete(box_name)
```

Comment:
I'm guessing this and create were left in for debug purposes? You can probably remove delete altogether. No reason for the user to be deleting the box anyway!